File: storage/data_transpiled/at_coder/abc027/B/4925032.java
public static int islandAndBridge ( @ Nonnegative int N , @ Nonnegative int [ ] A ) {
  if ( Integer . parseInt ( A [ 0 ] ) % N > 0 ) return - 1 ;
  int s = 0 ;
  int goal = Integer . parseInt ( A [ 0 ] ) / N ;
  int bridges = 0 ;
  for ( int a : A ) {
    s += a - goal ;
    if ( s != 0 ) {
      bridges ++ ;
    }
  }
  return bridges ;
}

File: storage/data_original/at_coder/abc027/B/4925032.py
def island_and_bridge(N: int, A: list) -> int:
    if sum(A) % N > 0:
        return - 1

    # s ??????????????????????????
    # ??? 0 ??????????????????
    s = 0
    goal = sum(A) // N
    bridges = 0
    for a in A:
        s += a - goal
        if s != 0:
            bridges += 1
    return bridges


if __name__ == "__main__":
    N = int(input())
    A = [int(s) for s in input().split()]
    ans = island_and_bridge(N, A)
    print(ans)